packages/salesforcedx-vscode-core/src/conflict/conflictDetectionService.ts

```diff
@@ -0,0 +1,248 @@
+    'conflicts'
+  );
+
+  private readonly relativePackageXmlPath = path.join(
+    this.relativeMetdataTempPath,
+    'package.xml'
+  );
+
+  private readonly relativeUnpackagedPath = path.join(
+    this.relativeMetdataTempPath,
+    'unpackaged'
+  );
+
+  private readonly relativeConvertedPath = path.join(
+    '.sfdx',
+    'tools',
+    'conflicts',
+    'converted'
+  );
+
+  public buildRetrieveOrgSourceCommand(data: ConflictDetectionConfig): Command {
+    return new SfdxCommandBuilder()
+      .withDescription(nls.localize('conflict_detect_retrieve_org_source'))
+      .withArg('force:mdapi:retrieve')
+      .withFlag('--retrievetargetdir', this.relativeMetdataTempPath)
+      .withFlag('--unpackaged', this.relativePackageXmlPath)
+      .withFlag('--targetusername', data.usernameOrAlias)
+      .withLogName('conflict_detect_retrieve_org_source')
+      .build();
+  }
+
+  public buildMetadataApiConvertOrgSourceCommand(
+    data: ConflictDetectionConfig
+  ): Command {
+    return new SfdxCommandBuilder()
+      .withDescription(nls.localize('conflict_detect_convert_org_source'))
+      .withArg('force:mdapi:convert')
+      .withFlag('--rootdir', this.relativeUnpackagedPath)
+      .withFlag('--outputdir', this.relativeConvertedPath)
+      .withLogName('conflict_detect_convert_org_source')
+      .build();
+  }
+
+  public async checkForConflicts(
+    data: ConflictDetectionConfig
+  ): Promise<DirectoryDiffResults> {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const results = await this.checkForConflictsInternal(
+      getRootWorkspacePath(),
+      data,
+      cancellationTokenSource,
+      cancellationToken
+    );
+    return results;
+  }
+
+  private async checkForConflictsInternal(
+    projectPath: string,
+    data: ConflictDetectionConfig,
+    cancellationTokenSource: any,
+    cancellationToken: any
+  ): Promise<DirectoryDiffResults> {
+    const projectMetadataTempPath = path.join(
+      projectPath,
+      this.relativeMetdataTempPath
+    );
+    const retrievePackageXmlPath = path.join(
+      projectPath,
+      this.relativePackageXmlPath
+    );
+    const remoteSourcePath: string = path.join(
+      projectPath,
+      this.relativeConvertedPath
+    );
+    const unpackagedZipFile = path.join(
+      projectMetadataTempPath,
+      'unpackaged.zip'
+    );
+    const localSourcePath: string = path.join(projectPath, data.packageDir);
+
+    // 1: create the shadow directory
+    try {
+      shell.mkdir('-p', projectMetadataTempPath);
+      shell.cp(data.manifest, retrievePackageXmlPath);
+    } catch (error) {
+      console.error(error);
+      channelService.appendLine(
+        nls.localize('error_creating_packagexml', error.toString())
+      );
+      notificationService.showErrorMessage(
+        nls.localize('error_creating_packagexml', error.toString())
+      );
+      return Promise.reject();
```

Comment:
Can we send telemetryExceptions in the different catch blocks of this method as well?

Reply:
Will do.

Reply:
Done